src/components/LanguageSwitch/LanguageSwitch.tsx

```diff
@@ -0,0 +1,157 @@
+  },
+  'de-de': {
+    translation: t('lng.german'),
+    icon: 'germany',
+  },
+  'es-es': {
+    translation: t('lng.spanish'),
+    icon: 'spain',
+  },
+  'zh-cn': {
+    translation: t('lng.chinese'),
+    icon: 'china',
+  },
+});
+
+const LanguageSwitch = () => {
+  const { t } = useTranslation();
+  const [open, setOpen] = React.useState(false);
+  const anchorRef = React.useRef<HTMLButtonElement>(null);
+
+  const languages = i18next.options.resources ? Object.keys(i18next.options.resources) : [];
+  const currentLanguage = i18next.language || i18next.options?.fallbackLng?.[0];
+
+  const { translation: userLanguage } = getTranslatedCurrentLanguage(t)[currentLanguage.toLowerCase()];
+
+  const handleToggle = () => {
+    setOpen(prevOpen => !prevOpen);
+  };
+
+  const handleClose = (event: MouseEvent<HTMLLIElement | Document | HTMLAnchorElement>) => {
+    if (anchorRef.current) {
+      if (anchorRef.current.contains(event.currentTarget)) {
+        return;
+      }
+    }
+
+    setOpen(false);
+  };
+
+  // return focus to the button when we transitioned from !open -> open
+  const prevOpen = React.useRef(open);
+  React.useEffect(() => {
+    if (prevOpen.current === true && open === false) {
+      if (anchorRef.current) {
+        anchorRef.current.focus();
+      }
+    }
+
+    prevOpen.current = open;
+  }, [open]);
+
+  const handleSwitchLanguage = (language: string) => (event: MouseEvent<HTMLLIElement>) => {
+    i18next.changeLanguage(language);
+    handleClose(event);
+  };
+
+  return (
+    <>
+      <Tooltip enterDelay={300} title={t('changeLanguage')}>
+        <SwitchButton color="inherit" onClick={handleToggle} ref={anchorRef}>
+          <LanguageIcon />
+          <span>{userLanguage}</span>
+          <ExpandMoreIcon fontSize="small" />
+        </SwitchButton>
+      </Tooltip>
+      <Popper anchorEl={anchorRef.current} disablePortal={true} open={open} role={undefined} transition={true}>
+        {({ TransitionProps }) => (
+          <Grow {...TransitionProps}>
+            <Paper>
+              <ClickAwayListener onClickAway={handleClose}>
+                <MenuList>
+                  {languages.map(language => {
+                    const { icon, translation } = getTranslatedCurrentLanguage(t)[language.toLowerCase()];
+                    return (
+                      <StyledMenuItem
+                        key={language}
+                        onClick={handleSwitchLanguage(language)}
+                        selected={userLanguage === translation}>
+                        <Icon name={icon} size="md" />
+                        {translation}
+                      </StyledMenuItem>
+                    );
+                  })}
+                  <Box my={1}>
+                    <Divider />
+                  </Box>
+                  <MenuItem button={true}>
+                    <StyledLink external={true} onClick={handleClose} to="https://github.com/verdaccio/ui#translations">
```

Comment:
I'd add this to a constant above, it is hardly visible here `https://github.com/verdaccio/ui#translations` . Furthermore, the README might change and this link will be broken. I'd rather remove the `#translations` part to be sure.